spring-cloud-skipper-server-core/src/main/java/org/springframework/cloud/skipper/server/deployer/strategies/HandleHealthCheckStep.java

```diff
@@ -0,0 +1,152 @@
+
+/**
+ * Responsible for taking action based on the health of the latest deployed release. If
+ * healthy, then delete applications from the previous release. Otherwise delete the
+ * latest deployed release. Delegates to {@link DeleteStep}.
+ *
+ * @author Mark Pollack
+ * @author Ilayaperumal Gopinathan
+ */
+public class HandleHealthCheckStep {
+
+	private final Logger logger = LoggerFactory.getLogger(HandleHealthCheckStep.class);
+
+	private final ReleaseRepository releaseRepository;
+
+	private final DeployerRepository deployerRepository;
+
+	private final AppDeployerDataRepository appDeployerDataRepository;
+
+	private final DeleteStep deleteStep;
+
+	private final HealthCheckProperties healthCheckProperties;
+
+	private ReleaseManager releaseManager;
+
+	public HandleHealthCheckStep(ReleaseRepository releaseRepository,
+			DeployerRepository deployerRepository,
+			AppDeployerDataRepository appDeployerDataRepository,
+			DeleteStep deleteStep,
+			HealthCheckProperties healthCheckProperties) {
+		this.releaseRepository = releaseRepository;
+		this.deployerRepository = deployerRepository;
+		this.appDeployerDataRepository = appDeployerDataRepository;
+		this.deleteStep = deleteStep;
+		this.healthCheckProperties = healthCheckProperties;
+	}
+
+	@Transactional
+	public void handleHealthCheck(boolean healthy, Release existingRelease,
+			List<String> applicationNamesToUpgrade,
+			Release replacingRelease) {
+		if (healthy) {
+			updateReplacingReleaseState(replacingRelease);
+			deleteExistingRelease(existingRelease, applicationNamesToUpgrade);
+		}
+		else {
+			deleteReplacingRelease(replacingRelease);
+		}
+	}
+
+	private void updateReplacingReleaseState(Release replacingRelease) {
+		// Update Status in DB
+		Status status = new Status();
+		status.setStatusCode(StatusCode.DEPLOYED);
+		replacingRelease.getInfo().setStatus(status);
+		replacingRelease.getInfo().setDescription("Upgrade complete");
+		this.releaseRepository.save(replacingRelease);
+		logger.info("Release {}-v{} has been DEPLOYED", replacingRelease.getName(),
+				replacingRelease.getVersion());
+		logger.info("Apps in release {}-v{} are healthy.", replacingRelease.getName(),
+				replacingRelease.getVersion());
+	}
+
+	private void deleteReplacingRelease(Release replacingRelease) {
+		try {
+			logger.error("New release " + replacingRelease.getName() + " was not detected as healthy after " +
+					this.healthCheckProperties.getTimeoutInMillis() + "milliseconds.  " +
+					"Keeping existing release, and Deleting apps of replacing release");
+			this.releaseManager.delete(replacingRelease);
+			Status status = new Status();
+			status.setStatusCode(StatusCode.FAILED);
+			replacingRelease.getInfo().setStatus(status);
+			replacingRelease.getInfo().setStatus(status);
+			replacingRelease.getInfo().setDescription("Did not detect apps in repalcing release as healthy after " +
+					this.healthCheckProperties.getSleepInMillis() + " ms.");
```

Comment:
I think this needs to be `getTimeoutInMillis() `

Reply:
yes!